core/log.class.inc.php

```diff
@@ -603,85 +606,115 @@ public static function Trace($sMessage, $sChannel = null, $aContext = array())
+			return;
+		}
+
+		static::$m_oFileLog->$sLevel($sMessage, $sChannel, $aContext);
+	}
+
+	/**
+	 * @throws \ConfigException if log wrongly configured
+	 * @uses GetMinLogLevel
+	 */
+	final public static function IsLogLevelEnabled(string $sLevel, string $sChannel): bool
+	{
 		$sMinLogLevel = self::GetMinLogLevel($sChannel);
 
-		if ($sMinLogLevel === false || $sMinLogLevel === 'false')
-		{
-			return;
+		if ($sMinLogLevel === false || $sMinLogLevel === 'false') {
+			return false;
 		}
-		if (is_string($sMinLogLevel))
-		{
-			if (! isset(self::$aLevelsPriority[$sMinLogLevel]))
-			{
-				throw new Exception("invalid configuration for log_level '{$sMinLogLevel}' is not within the list: ".implode(',', array_keys(self::$aLevelsPriority)));
-			}
-			elseif (self::$aLevelsPriority[$sLevel] < self::$aLevelsPriority[$sMinLogLevel])
-			{
-				//priority too low regarding the conf, do not log this
-				return;
-			}
+		if (!is_string($sMinLogLevel)) {
+			return false;
 		}
 
-		static::$m_oFileLog->$sLevel($sMessage, $sChannel, $aContext);
+		if (!isset(self::$aLevelsPriority[$sMinLogLevel])) {
+			throw new ConfigException("invalid configuration for log_level '{$sMinLogLevel}' is not within the list: ".implode(',', array_keys(self::$aLevelsPriority)));
+		} elseif (self::$aLevelsPriority[$sLevel] < self::$aLevelsPriority[$sMinLogLevel]) {
+			return false;
+		}
+
+		return true;
 	}
 
 	/**
-	 * @param $sChannel
+	 * @param string $sChannel
+	 *
+	 * @return string one of the LEVEL_* const value : the one configured it if exists, otherwise default log level for this channel
+	 *       Config can be done globally : `'log_level_min' => LogAPI::LEVEL_TRACE,`
+	 *       Or per channel : `'log_level_min' => ['InlineImage' => LogAPI::LEVEL_TRACE, 'UserRequest' => LogAPI::LEVEL_TRACE],`
 	 *
-	 * @return string one of the LEVEL_* const value
-	 * @uses \LogAPI::LEVEL_DEFAULT
+	 * @uses \LogAPI::GetConfig()
+	 * @uses \LogAPI::GetLevelDefault
 	 */
-	private static function GetMinLogLevel($sChannel)
+	protected static function GetMinLogLevel($sChannel)
 	{
-		$oConfig = (static::$m_oMockMetaModelConfig !== null) ? static::$m_oMockMetaModelConfig :  \MetaModel::GetConfig();
-		if (!$oConfig instanceof Config)
-		{
-			return static::LEVEL_DEFAULT;
+		$oConfig = static::GetConfig();
+		if (!$oConfig instanceof Config) {
+			return static::GetLevelDefault();
 		}
 
 		$sLogLevelMin = $oConfig->Get('log_level_min');
 
-		if (empty($sLogLevelMin))
-		{
-			return static::LEVEL_DEFAULT;
+		if (empty($sLogLevelMin)) {
+			return static::GetLevelDefault();
 		}
 
-		if (!is_array($sLogLevelMin))
-		{
+		if (!is_array($sLogLevelMin)) {
 			return $sLogLevelMin;
 		}
 
-		if (isset($sLogLevelMin[$sChannel]))
-		{
+		if (isset($sLogLevelMin[$sChannel])) {
 			return $sLogLevelMin[$sChannel];
 		}
 
-		if (isset($sLogLevelMin[static::CHANNEL_DEFAULT]))
-		{
+		if (isset($sLogLevelMin[static::CHANNEL_DEFAULT])) {
 			return $sLogLevelMin[$sChannel];
```

Comment:
Is this even correct? If I'm correct, `$sLogLevelMin[$sChannel]` would not exist.
```suggestion
			return $sLogLevelMin[static::CHANNEL_DEFAULT]];
```

Reply:
It is not, you're right !
This is part of the first bug I created : N°3944. This will be discussed with the products team and fixed independently, so I'm closing this conversation.